includes/admin/class-wp-job-manager-settings.php

```diff
@@ -995,68 +1032,24 @@ class="regular-text settings-role-select"
 	}
 
 	/**
-	 * Role settings should be saved as a comma-separated list.
+	 * Sanitize the options related to capabilities, making the necessary conversions
+	 *
+	 * @param string[]|string $value
+	 * @return string[]
 	 */
-	public function pre_process_settings_save() {
-		$screen = get_current_screen();
-
-		if ( ! $screen || 'options' !== $screen->id ) {
-			return;
-		}
-
-		// phpcs:ignore WordPress.Security.NonceVerification.Missing -- Settings save will handle the nonce check.
-		if ( ! isset( $_POST['option_page'] ) || 'job_manager' !== $_POST['option_page'] ) {
-			return;
+	public function sanitize_capabilities( $value ) {
+		$value = wp_unslash( $value );
+		if ( is_string( $value ) ) {
+			$value = explode( ',', $value );
 		}
-
-		$capabilities_fields = [
-			'job_manager_browse_job_listings_capability',
-			'job_manager_view_job_listing_capability',
-		];
-		foreach ( $capabilities_fields as $capabilities_field ) {
-			// phpcs:disable WordPress.Security.NonceVerification.Missing -- Settings save will handle the nonce check.
-			if ( isset( $_POST[ $capabilities_field ] ) ) {
-				// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- Sanitized by `WP_Resume_Manager_Settings::capabilities_array_to_string()`
-				$input_capabilities_field_value = wp_unslash( $_POST[ $capabilities_field ] );
-				if ( is_array( $input_capabilities_field_value ) ) {
-					$_POST[ $capabilities_field ] = self::capabilities_array_to_string( $input_capabilities_field_value );
-				}
+		$result = [];
+		foreach ( $value as $item ) {
+			$item = strtolower( trim( sanitize_text_field( $item ) ) );
```

Comment:
I don't think `strtolower` is safe in this case. Custom roles can be [arbitrary strings](https://developer.wordpress.org/reference/classes/wp_roles/add_role/). Even though the default roles are lower case, I don't think they have to be.